presto-main/src/main/java/io/prestosql/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -858,24 +858,39 @@ protected Type visitFunctionCall(FunctionCall node, StackableAstVisitorContext<C
             }
 
             List<TypeSignatureProvider> argumentTypes = getCallArgumentTypes(node.getArguments(), context);
-            Signature function = resolveFunction(node, argumentTypes, metadata);
+
+            Signature function;
+            try {
+                function = metadata.resolveFunction(node.getName(), argumentTypes);
+            }
+            catch (PrestoException e) {
+                if (e.getLocation().isPresent()) {
```

Comment:
No. This is a tricky one. `metadata.resolveFunction` can fail either because the function doesn't exist (which throws a custom PrestoException w/o a location, or because the analysis of `argumentTypes` fails. The `argumentTypes` are analyzed lazily to deal with lambda expressions. In that case, we just want to re-throw the original exception.

Reply:
Ah, I didn't realize `TypeSignatureProvider` was lazy. So it does indeed have AST nodes inside.